src/main/java/com/atlassian/db/replica/spi/TransactionHook.java

```diff
@@ -0,0 +1,23 @@
+package com.atlassian.db.replica.spi;
+
+import java.sql.Connection;
+
+public interface TransactionHook {
+    /**
+     * Informs that {@code main} received an UPDATE, INSERT or DELETE or transaction commit
+     * when in a transaction.
+     *
+     * @param main connects to the main database
+     */
+    void write(Connection main);
```

Comment:
The past tense of "received an UPDATE" suggests this is called after the write.

```suggestion
    void postWrite(Connection main);
```